src/app/shared/services/translation.service.ts

```diff
@@ -67,57 +67,60 @@ setIndex() {
 }
 
 translate(text: string) {
-    if (typeof text !== 'string') {
-        return text;
-    }
-
-    const length = this.dictionary.value.length;
-
-    return this.binarySearch(this.dictionary.value, text.toLowerCase(), 0, length) || text;
-
+  if (typeof text !== 'string') {
+      return text;
+  }
+
+  const length = this.dictionary.value.length;
+
+  if (this.currentLanguage === 'hi') {
+      return this.binarySearch(this.dictionary.value, text.toLowerCase(), 0, length, 'translation') || text;
+  } else if (this.currentLanguage === 'or') {
+      return this.binarySearch(this.dictionary.value, text.toLowerCase(), 0, length, 'translation2') || text;
+  } else {
+      return text;
+  }
 }
 
-
-
-binarySearch(arr: Array<Dictionary>, text: string, start: number, end: number) {
-
-    const mid = Math.floor((start + end) / 2);
-
-    if (start > end) {
-        return null;
-    }
-
-    const startText = arr[start] && arr[start].text.toLowerCase();
-    const midText = arr[mid] && arr[mid].text.toLowerCase();
-    const endText = arr[end] && arr[end].text.toLowerCase();
-
-    if (start === end) {
-        if (startText === text) {
-            return arr[start].translation;
-        } else {
-            return null;
-        }
-    }
-
-    if (startText === text) {
-        return arr[start].translation;
-    }
-
-    if (endText === text) {
-        return arr[end].translation;
-    }
-
-    if (midText === text) {
-        return arr[mid].translation;
-    }
-
-    if (midText > text) {
-        return this.binarySearch(arr, text, start, mid - 1);
-    } else {
-        return this.binarySearch(arr, text, mid + 1, end);
-    }
+binarySearch(arr: Array<Dictionary>, text: string, start: number, end: number, field: string) {
```

Comment:
This code is wrongly formatted. The rest of the file is using indents of 4 spaces